Asterisk.2013/Asterisk.NET/Manager/ManagerConnection.cs

```diff
@@ -15,1566 +15,918 @@
+            Helper.RegisterEventHandler(registeredEventHandlers, typeof(DTMFEvent), arg => fireEvent(DTMF, arg));
+
+            Helper.RegisterEventHandler(registeredEventHandlers, typeof(VarSetEvent), arg => fireEvent(VarSet, arg));
+            Helper.RegisterEventHandler(registeredEventHandlers, typeof(AGIExecEvent), arg => fireEvent(AGIExec, arg));
+
+            Helper.RegisterEventHandler(registeredEventHandlers, typeof(ConfbridgeStartEvent), arg => fireEvent(ConfbridgeStart, arg));
+            Helper.RegisterEventHandler(registeredEventHandlers, typeof(ConfbridgeJoinEvent), arg => fireEvent(ConfbridgeJoin, arg));
+            Helper.RegisterEventHandler(registeredEventHandlers, typeof(ConfbridgeLeaveEvent), arg => fireEvent(ConfbridgeLeave, arg));
+            Helper.RegisterEventHandler(registeredEventHandlers, typeof(ConfbridgeEndEvent), arg => fireEvent(ConfbridgeEnd, arg));
+            Helper.RegisterEventHandler(registeredEventHandlers, typeof(ConfbridgeTalkingEvent), arg => fireEvent(ConfbridgeTalking, arg));
+
+            Helper.RegisterEventHandler(registeredEventHandlers, typeof(FailedACLEvent), arg => fireEvent(FailedACL, arg));
+
+            Helper.RegisterEventHandler(registeredEventHandlers, typeof(AttendedTransferEvent), arg => fireEvent(AttendedTransfer, arg));
+            Helper.RegisterEventHandler(registeredEventHandlers, typeof(BridgeCreateEvent), arg => fireEvent(BridgeCreate, arg));
+            Helper.RegisterEventHandler(registeredEventHandlers, typeof(BridgeDestroyEvent), arg => fireEvent(BridgeDestroy, arg));
+            Helper.RegisterEventHandler(registeredEventHandlers, typeof(BridgeEnterEvent), arg => fireEvent(BridgeEnter, arg));
+            Helper.RegisterEventHandler(registeredEventHandlers, typeof(BridgeLeaveEvent), arg => fireEvent(BridgeLeave, arg));
+            Helper.RegisterEventHandler(registeredEventHandlers, typeof(BlindTransferEvent), arg => fireEvent(BlindTransfer, arg));
+            Helper.RegisterEventHandler(registeredEventHandlers, typeof(DialBeginEvent), arg => fireEvent(DialBegin, arg));
+            Helper.RegisterEventHandler(registeredEventHandlers, typeof(DialEndEvent), arg => fireEvent(DialEnd, arg));
+            Helper.RegisterEventHandler(registeredEventHandlers, typeof(QueueCallerJoinEvent), arg => fireEvent(QueueCallerJoin, arg));
+            Helper.RegisterEventHandler(registeredEventHandlers, typeof(QueueCallerLeaveEvent), arg => fireEvent(QueueCallerLeave, arg));
+            Helper.RegisterEventHandler(registeredEventHandlers, typeof(QueueMemberPauseEvent), arg => fireEvent(QueueMemberPause, arg));
 
             #endregion
 
-            this.internalEvent += new ManagerEventHandler(internalEventHandler);
-		}
-		#endregion
-
-		#region Constructor - ManagerConnection(hostname, port, username, password)
-		/// <summary>
-		/// Creates a new instance with the given connection parameters.
-
```

Comment:
You need to call `return` after this line, or else `fireAllEvents` will fire an event when the event is already handled. The `fireAllEvents` is only suppose to happen if the event is unhandled, check the [documentation](http://asternet.github.io/AsterNET/html/P_AsterNET_Manager_ManagerConnection_FireAllEvents.htm).
See: #121

Reply:
Sorry, but why Line 633? Maybe i need to call return after 
<a href="https://github.com/AsterNET/AsterNET/pull/93/files/d1bb4249b88e3bbe3b1e897bf335071d5335887a#diff-d04fbe86bbb901c0836aa4957c1406ecR637">Line 637</a>. Because in the previous version (before my change) we called <code>break</code> after all<code>case</code>, and return after <code>default</code>

Reply:
You are partially right.
You also need to return after [line 637](https://github.com/AsterNET/AsterNET/pull/93/files/d1bb4249b88e3bbe3b1e897bf335071d5335887a#diff-d04fbe86bbb901c0836aa4957c1406ecR637).

If you don't return those two lines (633 & 637), the if statement at [line 640](https://github.com/AsterNET/AsterNET/pull/93/files/d1bb4249b88e3bbe3b1e897bf335071d5335887a#diff-d04fbe86bbb901c0836aa4957c1406ecR640) will fire another event even if an event was already fired.